packages/playground/src/context/messages/reducer.ts

```diff
@@ -0,0 +1,171 @@
+import { CompilationResult } from '@paritytech/commontypes';
+import { Message, Status, Severity, Prompt } from '@paritytech/components/';
+
+export const defaultState: MessageState = {
+  messages: [],
+  nextId: 0,
+};
+
+export type MessageState = {
+  messages: Array<Message>;
+  nextId: number;
+};
+
+export type Action = {
+  type: 'LOG_COMPILE' | 'LOG_SYSTEM' | 'LOG_GIST';
+  payload: {
+    status: Status;
+    content: string;
+    result?: CompilationResult;
+  };
+};
+
+export type MessageDispatch = (action: Action) => void;
+
+const lastId = (state: MessageState, prompt: Prompt): number => {
+  const arr = state.messages.filter(message => message.prompt === prompt);
+  // if no last id available, return nextId, even though it should not happen
+  if (!arr[arr.length - 1]) return state.nextId;
+  return arr[arr.length - 1].id;
+};
+
+export const reducer = (state: MessageState, { type, payload }: Action): MessageState => {
+  switch (type) {
+    case 'LOG_SYSTEM':
+      if (payload.status === 'IN_PROGRESS') {
```

Comment:
the if/else if's.... couldn't we just use a switch statement?

Reply:
it would be more readable.

Reply:
I refactored the reducer and added switch statements when there were more than two cases.